corehq/apps/reports/standard/deployments.py

```diff
@@ -781,8 +781,18 @@ def _pct(val, total):
         query = self.user_query().run()
 
         aggregations = query.aggregations
-        last_submission_buckets = aggregations[0].normalized_buckets
-        last_sync_buckets = aggregations[1].normalized_buckets
+        if self.selected_app_id:
+            last_submission_buckets = (aggregations[0]
+                                       .filtered_last_submissions
+                                       .last_submissions_date_histogram
+                                       .normalized_buckets)
```

Comment:
same as the other comment, can you help understand why this is different for if there is a selected app id?

Reply:
The aggregations we used are all nested inside each other in the following hierarchy (nested/filter/histogram). When creating these aggregations we assigned them each a name (first input param for the aggregation class). These names are where the results of the relevant aggregation will reside in `aggregations` of the ES query. Since the aggregations are nested inside each other, we have to drill down into each of them to eventually get to the buckets for the histogram.

TL:DR This is different because we make use of a nested and filter aggregation. Since all the aggregations are nested inside each other, we need to drill down all the aggregations to get to the buckets for the histogram.

Reply:
okay, what are `filtered_last_submissions` & `last_submissions_date_histogram`?

Reply:
The `filtered_last_submissions` is the `FilterAggregation` and `last_submissions_date_histogram` is the `DateHistogram` that we added. Since the histogram aggregation is nested inside the filter aggregation which is nested inside the nested aggregation, we have to drilldown through the nesting to reach the histogram with `NestedAggregation.FilterAggregation.HistogramAggregation.normalized_buckets`

Reply:
> The filtered_last_submissions is the FilterAggregation and last_submissions_date_histogram is the DateHistogram that we added

Ah, it is coming from the dynamically generated naming we did.
This is confusing here because it looks like a property being referred when it is not, so searching for `filtered_last_submissions` and `last_submissions_date_histogram` in code does not give anything.

Can that be converted to a dict fetch like 
`aggregations[0]['filtered_last_submissions']`? Or could think of better ways like calling `_get_histogram_aggregation_for_app` with `filtered_last_submissions` and `last_submissions_date_histogram` or just leave a comment.

this is a non blocking comment.

Reply:
Given that we use dot notation throughout HQ to refer to named aggregations (e.g. [here](https://github.com/dimagi/commcare-hq/blob/1eae8fc1cc3ae3c267b6501343f4a9ae348b6da9/corehq/apps/app_manager/dbaccessors.py#L534) and [here](https://github.com/dimagi/commcare-hq/blob/1eae8fc1cc3ae3c267b6501343f4a9ae348b6da9/corehq/apps/reports/analytics/esaccessors.py#L64)) I'm opted to rather not convert it to a dict fetch and keep it consistent with the rest of HQ. 

I'm happy to add a comment however, and explain where the references are coming from.

Reply:
Thanks @zandre-eng , fine either way.

Reply:
@mkangia A comment has been added ([5a6c37a](https://github.com/dimagi/commcare-hq/pull/34178/commits/5a6c37a3abda8a8cee7217643ea15db89652f833)).

Reply:
I haven't fully thought this through but my on the fly 2c is that the dynamic naming isn't necessary and you could do away with it entirely, and just use the names passed in to the `_get_histogram_aggregation_for_app`